### ISSUE TITLE
Typed tagged template literals trigger unexpected "no-mixed-operators" warning

### ISSUE THREAD

**Issue by jazzfog** [eslint/eslint#11571]
**Tell us about your environment**

* **ESLint:** `5.12.0`
* **TypeScript:** `3.3.3333`
* **Node:** `11.13.0`
* **npm:** `6.7.0`
* **yarn:** `1.15.2`
* **create-react-app:** `Windows 10`
* **styled-components:** `4.2.0`
* **@types/styled-components:** `4.1.12`
* **OS Version:** `Windows 10`

**What parser (default, Babel-ESLint, etc.) are you using?**
What `create-react-app` uses

**Please show your full configuration:**
Default `create-react-app`'s configuration

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import React, { FunctionComponent } from 'react';

import styled from 'styled-components'

type TButtonProps = {
  selected: boolean;
};

const ButtonStyle = styled.button<TButtonProps>`
  font-weight: ${props => props.selected ? 'bold' : 'normal'};
`;

type TProps = {
  text: string;
}

export const Button: FunctionComponent<TProps> = (props) => {
  return <ButtonStyle selected={false}>{props.text}</ButtonStyle>;
};
```

```bash
eslint ./src/Button.tsx
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
Errors reported in this line
```
const ButtonStyle = styled.button<TButtonProps>`
```
ESlint output:
```
$ eslint ./src/Button.tsx

d:\web\project-folder\front\src\Button.tsx
  9:34  warning  Unexpected mix of '<' and '>'  no-mixed-operators
  9:47  warning  Unexpected mix of '<' and '>'  no-mixed-operators

2 problems (0 errors, 2 warnings)

Done in 0.98s.
```

As I understand the way I type the style for my component is legitimate and it is [suggested by the styled-components documentation](https://www.styled-components.com/docs/api#using-custom-props).

**Are you willing to submit a pull request to fix this bug?**
No

**Keywords**
`TypeScript` `Styled-Components` `Eslint` `no-mixed-operators` `create-react-app`.

**Comment by not-an-aardvark** [comment]
Hi, thanks for reporting an issue. It seems like the parser is interpreting the type argument `<TProps>` as a less-than and greater-than operator. With the template string, this interpretation doesn't result in any syntax errors.

In other words, the code is getting interpreted like this (parentheses/spacing added):

```js
const ButtonStyle = (styled.button < TButtonProps) > (`
  font-weight: ${props => props.selected ? 'bold' : 'normal'};
`);
```

This seems like a bug in the parser you're using. What is the output of `eslint --print-config d:\web\project-folder\front\src\Button.tsx`? (I'm specifically interested in the `parser` property of the output.)

**Comment by jazzfog** [comment]
Sure, full output:
```
$ eslint --print-config src/Button.tsx
{
  "globals": {
    "AbortController": false,
    "AbortSignal": false,
    "addEventListener": false,
    "alert": false,
    "AnalyserNode": false,
    "Animation": false,
    "AnimationEffectReadOnly": false,
    "AnimationEffectTiming": false,
    "AnimationEffectTimingReadOnly": false,
    "AnimationEvent": false,
    "AnimationPlaybackEvent": false,
    "AnimationTimeline": false,
    "applicationCache": false,
    "ApplicationCache": false,
    "ApplicationCacheErrorEvent": false,
    "atob": false,
    "Attr": false,
    "Audio": false,
    "AudioBuffer": false,
    "AudioBufferSourceNode": false,
    "AudioContext": false,
    "AudioDestinationNode": false,
    "AudioListener": false,
    "AudioNode": false,
    "AudioParam": false,
    "AudioProcessingEvent": false,
    "AudioScheduledSourceNode": false,
    "AudioWorkletGlobalScope ": false,
    "AudioWorkletNode": false,
    "AudioWorkletProcessor": false,
    "BarProp": false,
    "BaseAudioContext": false,
    "BatteryManager": false,
    "BeforeUnloadEvent": false,
    "BiquadFilterNode": false,
    "Blob": false,
    "BlobEvent": false,
    "blur": false,
    "BroadcastChannel": false,
    "btoa": false,
    "BudgetService": false,
    "ByteLengthQueuingStrategy": false,
    "Cache": false,
    "caches": false,
    "CacheStorage": false,
    "cancelAnimationFrame": false,
    "cancelIdleCallback": false,
    "CanvasCaptureMediaStreamTrack": false,
    "CanvasGradient": false,
    "CanvasPattern": false,
    "CanvasRenderingContext2D": false,
    "ChannelMergerNode": false,
    "ChannelSplitterNode": false,
    "CharacterData": false,
    "clearInterval": false,
    "clearTimeout": false,
    "clientInformation": false,
    "ClipboardEvent": false,
    "close": false,
    "closed": false,
    "CloseEvent": false,
    "Comment": false,
    "CompositionEvent": false,
    "confirm": false,
    "console": false,
    "ConstantSourceNode": false,
    "ConvolverNode": false,
    "CountQueuingStrategy": false,
    "createImageBitmap": false,
    "Credential": false,
    "CredentialsContainer": false,
    "crypto": false,
    "Crypto": false,
    "CryptoKey": false,
    "CSS": false,
    "CSSConditionRule": false,
    "CSSFontFaceRule": false,
    "CSSGroupingRule": false,
    "CSSImportRule": false,
    "CSSKeyframeRule": false,
    "CSSKeyframesRule": false,
    "CSSMediaRule": false,
    "CSSNamespaceRule": false,
    "CSSPageRule": false,
    "CSSRule": false,
    "CSSRuleList": false,
    "CSSStyleDeclaration": false,
    "CSSStyleRule": false,
    "CSSStyleSheet": false,
    "CSSSupportsRule": false,
    "CustomElementRegistry": false,
    "customElements": false,
    "CustomEvent": false,
    "DataTransfer": false,
    "DataTransferItem": false,
    "DataTransferItemList": false,
    "defaultstatus": false,
    "defaultStatus": false,
    "DelayNode": false,
    "DeviceMotionEvent": false,
    "DeviceOrientationEvent": false,
    "devicePixelRatio": false,
    "dispatchEvent": false,
    "document": false,
    "Document": false,
    "DocumentFragment": false,
    "DocumentType": false,
    "DOMError": false,
    "DOMException": false,
    "DOMImplementation": false,
    "DOMMatrix": false,
    "DOMMatrixReadOnly": false,
    "DOMParser": false,
    "DOMPoint": false,
    "DOMPointReadOnly": false,
    "DOMQuad": false,
    "DOMRect": false,
    "DOMRectReadOnly": false,
    "DOMStringList": false,
    "DOMStringMap": false,
    "DOMTokenList": false,
    "DragEvent": false,
    "DynamicsCompressorNode": false,
    "Element": false,
    "ErrorEvent": false,
    "event": false,
    "Event": false,
    "EventSource": false,
    "EventTarget": false,
    "external": false,
    "fetch": false,
    "File": false,
    "FileList": false,
    "FileReader": false,
    "find": false,
    "focus": false,
    "FocusEvent": false,
    "FontFace": false,
    "FontFaceSetLoadEvent": false,
    "FormData": false,
    "frameElement": false,
    "frames": false,
    "GainNode": false,
    "Gamepad": false,
    "GamepadButton": false,
    "GamepadEvent": false,
    "getComputedStyle": false,
    "getSelection": false,
    "HashChangeEvent": false,
    "Headers": false,
    "history": false,
    "History": false,
    "HTMLAllCollection": false,
    "HTMLAnchorElement": false,
    "HTMLAreaElement": false,
    "HTMLAudioElement": false,
    "HTMLBaseElement": false,
    "HTMLBodyElement": false,
    "HTMLBRElement": false,
    "HTMLButtonElement": false,
    "HTMLCanvasElement": false,
    "HTMLCollection": false,
    "HTMLContentElement": false,
    "HTMLDataElement": false,
    "HTMLDataListElement": false,
    "HTMLDetailsElement": false,
    "HTMLDialogElement": false,
    "HTMLDirectoryElement": false,
    "HTMLDivElement": false,
    "HTMLDListElement": false,
    "HTMLDocument": false,
    "HTMLElement": false,
    "HTMLEmbedElement": false,
    "HTMLFieldSetElement": false,
    "HTMLFontElement": false,
    "HTMLFormControlsCollection": false,
    "HTMLFormElement": false,
    "HTMLFrameElement": false,
    "HTMLFrameSetElement": false,
    "HTMLHeadElement": false,
    "HTMLHeadingElement": false,
    "HTMLHRElement": false,
    "HTMLHtmlElement": false,
    "HTMLIFrameElement": false,
    "HTMLImageElement": false,
    "HTMLInputElement": false,
    "HTMLLabelElement": false,
    "HTMLLegendElement": false,
    "HTMLLIElement": false,
    "HTMLLinkElement": false,
    "HTMLMapElement": false,
    "HTMLMarqueeElement": false,
    "HTMLMediaElement": false,
    "HTMLMenuElement": false,
    "HTMLMetaElement": false,
    "HTMLMeterElement": false,
    "HTMLModElement": false,
    "HTMLObjectElement": false,
    "HTMLOListElement": false,
    "HTMLOptGroupElement": false,
    "HTMLOptionElement": false,
    "HTMLOptionsCollection": false,
    "HTMLOutputElement": false,
    "HTMLParagraphElement": false,
    "HTMLParamElement": false,
    "HTMLPictureElement": false,
    "HTMLPreElement": false,
    "HTMLProgressElement": false,
    "HTMLQuoteElement": false,
    "HTMLScriptElement": false,
    "HTMLSelectElement": false,
    "HTMLShadowElement": false,
    "HTMLSlotElement": false,
    "HTMLSourceElement": false,
    "HTMLSpanElement": false,
    "HTMLStyleElement": false,
    "HTMLTableCaptionElement": false,
    "HTMLTableCellElement": false,
    "HTMLTableColElement": false,
    "HTMLTableElement": false,
    "HTMLTableRowElement": false,
    "HTMLTableSectionElement": false,
    "HTMLTemplateElement": false,
    "HTMLTextAreaElement": false,
    "HTMLTimeElement": false,
    "HTMLTitleElement": false,
    "HTMLTrackElement": false,
    "HTMLUListElement": false,
    "HTMLUnknownElement": false,
    "HTMLVideoElement": false,
    "IDBCursor": false,
    "IDBCursorWithValue": false,
    "IDBDatabase": false,
    "IDBFactory": false,
    "IDBIndex": false,
    "IDBKeyRange": false,
    "IDBObjectStore": false,
    "IDBOpenDBRequest": false,
    "IDBRequest": false,
    "IDBTransaction": false,
    "IDBVersionChangeEvent": false,
    "IdleDeadline": false,
    "IIRFilterNode": false,
    "Image": false,
    "ImageBitmap": false,
    "ImageBitmapRenderingContext": false,
    "ImageCapture": false,
    "ImageData": false,
    "indexedDB": false,
    "innerHeight": false,
    "innerWidth": false,
    "InputEvent": false,
    "IntersectionObserver": false,
    "IntersectionObserverEntry": false,
    "Intl": false,
    "isSecureContext": false,
    "KeyboardEvent": false,
    "KeyframeEffect": false,
    "KeyframeEffectReadOnly": false,
    "length": false,
    "localStorage": false,
    "location": true,
    "Location": false,
    "locationbar": false,
    "matchMedia": false,
    "MediaDeviceInfo": false,
    "MediaDevices": false,
    "MediaElementAudioSourceNode": false,
    "MediaEncryptedEvent": false,
    "MediaError": false,
    "MediaKeyMessageEvent": false,
    "MediaKeySession": false,
    "MediaKeyStatusMap": false,
    "MediaKeySystemAccess": false,
    "MediaList": false,
    "MediaQueryList": false,
    "MediaQueryListEvent": false,
    "MediaRecorder": false,
    "MediaSettingsRange": false,
    "MediaSource": false,
    "MediaStream": false,
    "MediaStreamAudioDestinationNode": false,
    "MediaStreamAudioSourceNode": false,
    "MediaStreamEvent": false,
    "MediaStreamTrack": false,
    "MediaStreamTrackEvent": false,
    "menubar": false,
    "MessageChannel": false,
    "MessageEvent": false,
    "MessagePort": false,
    "MIDIAccess": false,
    "MIDIConnectionEvent": false,
    "MIDIInput": false,
    "MIDIInputMap": false,
    "MIDIMessageEvent": false,
    "MIDIOutput": false,
    "MIDIOutputMap": false,
    "MIDIPort": false,
    "MimeType": false,
    "MimeTypeArray": false,
    "MouseEvent": false,
    "moveBy": false,
    "moveTo": false,
    "MutationEvent": false,
    "MutationObserver": false,
    "MutationRecord": false,
    "name": false,
    "NamedNodeMap": false,
    "NavigationPreloadManager": false,
    "navigator": false,
    "Navigator": false,
    "NetworkInformation": false,
    "Node": false,
    "NodeFilter": false,
    "NodeIterator": false,
    "NodeList": false,
    "Notification": false,
    "OfflineAudioCompletionEvent": false,
    "OfflineAudioContext": false,
    "offscreenBuffering": false,
    "OffscreenCanvas": true,
    "onabort": true,
    "onafterprint": true,
    "onanimationend": true,
    "onanimationiteration": true,
    "onanimationstart": true,
    "onappinstalled": true,
    "onauxclick": true,
    "onbeforeinstallprompt": true,
    "onbeforeprint": true,
    "onbeforeunload": true,
    "onblur": true,
    "oncancel": true,
    "oncanplay": true,
    "oncanplaythrough": true,
    "onchange": true,
    "onclick": true,
    "onclose": true,
    "oncontextmenu": true,
    "oncuechange": true,
    "ondblclick": true,
    "ondevicemotion": true,
    "ondeviceorientation": true,
    "ondeviceorientationabsolute": true,
    "ondrag": true,
    "ondragend": true,
    "ondragenter": true,
    "ondragleave": true,
    "ondragover": true,
    "ondragstart": true,
    "ondrop": true,
    "ondurationchange": true,
    "onemptied": true,
    "onended": true,
    "onerror": true,
    "onfocus": true,
    "ongotpointercapture": true,
    "onhashchange": true,
    "oninput": true,
    "oninvalid": true,
    "onkeydown": true,
    "onkeypress": true,
    "onkeyup": true,
    "onlanguagechange": true,
    "onload": true,
    "onloadeddata": true,
    "onloadedmetadata": true,
    "onloadstart": true,
    "onlostpointercapture": true,
    "onmessage": true,
    "onmessageerror": true,
    "onmousedown": true,
    "onmouseenter": true,
    "onmouseleave": true,
    "onmousemove": true,
    "onmouseout": true,
    "onmouseover": true,
    "onmouseup": true,
    "onmousewheel": true,
    "onoffline": true,
    "ononline": true,
    "onpagehide": true,
    "onpageshow": true,
    "onpause": true,
    "onplay": true,
    "onplaying": true,
    "onpointercancel": true,
    "onpointerdown": true,
    "onpointerenter": true,
    "onpointerleave": true,
    "onpointermove": true,
    "onpointerout": true,
    "onpointerover": true,
    "onpointerup": true,
    "onpopstate": true,
    "onprogress": true,
    "onratechange": true,
    "onrejectionhandled": true,
    "onreset": true,
    "onresize": true,
    "onscroll": true,
    "onsearch": true,
    "onseeked": true,
    "onseeking": true,
    "onselect": true,
    "onstalled": true,
    "onstorage": true,
    "onsubmit": true,
    "onsuspend": true,
    "ontimeupdate": true,
    "ontoggle": true,
    "ontransitionend": true,
    "onunhandledrejection": true,
    "onunload": true,
    "onvolumechange": true,
    "onwaiting": true,
    "onwheel": true,
    "open": false,
    "openDatabase": false,
    "opener": false,
    "Option": false,
    "origin": false,
    "OscillatorNode": false,
    "outerHeight": false,
    "outerWidth": false,
    "PageTransitionEvent": false,
    "pageXOffset": false,
    "pageYOffset": false,
    "PannerNode": false,
    "parent": false,
    "Path2D": false,
    "PaymentAddress": false,
    "PaymentRequest": false,
    "PaymentRequestUpdateEvent": false,
    "PaymentResponse": false,
    "performance": false,
    "Performance": false,
    "PerformanceEntry": false,
    "PerformanceLongTaskTiming": false,
    "PerformanceMark": false,
    "PerformanceMeasure": false,
    "PerformanceNavigation": false,
    "PerformanceNavigationTiming": false,
    "PerformanceObserver": false,
    "PerformanceObserverEntryList": false,
    "PerformancePaintTiming": false,
    "PerformanceResourceTiming": false,
    "PerformanceTiming": false,
    "PeriodicWave": false,
    "Permissions": false,
    "PermissionStatus": false,
    "personalbar": false,
    "PhotoCapabilities": false,
    "Plugin": false,
    "PluginArray": false,
    "PointerEvent": false,
    "PopStateEvent": false,
    "postMessage": false,
    "Presentation": false,
    "PresentationAvailability": false,
    "PresentationConnection": false,
    "PresentationConnectionAvailableEvent": false,
    "PresentationConnectionCloseEvent": false,
    "PresentationConnectionList": false,
    "PresentationReceiver": false,
    "PresentationRequest": false,
    "print": false,
    "ProcessingInstruction": false,
    "ProgressEvent": false,
    "PromiseRejectionEvent": false,
    "prompt": false,
    "PushManager": false,
    "PushSubscription": false,
    "PushSubscriptionOptions": false,
    "queueMicrotask": false,
    "RadioNodeList": false,
    "Range": false,
    "ReadableStream": false,
    "registerProcessor": false,
    "RemotePlayback": false,
    "removeEventListener": false,
    "Request": false,
    "requestAnimationFrame": false,
    "requestIdleCallback": false,
    "resizeBy": false,
    "ResizeObserver": false,
    "ResizeObserverEntry": false,
    "resizeTo": false,
    "Response": false,
    "RTCCertificate": false,
    "RTCDataChannel": false,
    "RTCDataChannelEvent": false,
    "RTCDtlsTransport": false,
    "RTCIceCandidate": false,
    "RTCIceGatherer": false,
    "RTCIceTransport": false,
    "RTCPeerConnection": false,
    "RTCPeerConnectionIceEvent": false,
    "RTCRtpContributingSource": false,
    "RTCRtpReceiver": false,
    "RTCRtpSender": false,
    "RTCSctpTransport": false,
    "RTCSessionDescription": false,
    "RTCStatsReport": false,
    "RTCTrackEvent": false,
    "screen": false,
    "Screen": false,
    "screenLeft": false,
    "ScreenOrientation": false,
    "screenTop": false,
    "screenX": false,
    "screenY": false,
    "ScriptProcessorNode": false,
    "scroll": false,
    "scrollbars": false,
    "scrollBy": false,
    "scrollTo": false,
    "scrollX": false,
    "scrollY": false,
    "SecurityPolicyViolationEvent": false,
    "Selection": false,
    "self": false,
    "ServiceWorker": false,
    "ServiceWorkerContainer": false,
    "ServiceWorkerRegistration": false,
    "sessionStorage": false,
    "setInterval": false,
    "setTimeout": false,
    "ShadowRoot": false,
    "SharedWorker": false,
    "SourceBuffer": false,
    "SourceBufferList": false,
    "speechSynthesis": false,
    "SpeechSynthesisEvent": false,
    "SpeechSynthesisUtterance": false,
    "StaticRange": false,
    "status": false,
    "statusbar": false,
    "StereoPannerNode": false,
    "stop": false,
    "Storage": false,
    "StorageEvent": false,
    "StorageManager": false,
    "styleMedia": false,
    "StyleSheet": false,
    "StyleSheetList": false,
    "SubtleCrypto": false,
    "SVGAElement": false,
    "SVGAngle": false,
    "SVGAnimatedAngle": false,
    "SVGAnimatedBoolean": false,
    "SVGAnimatedEnumeration": false,
    "SVGAnimatedInteger": false,
    "SVGAnimatedLength": false,
    "SVGAnimatedLengthList": false,
    "SVGAnimatedNumber": false,
    "SVGAnimatedNumberList": false,
    "SVGAnimatedPreserveAspectRatio": false,
    "SVGAnimatedRect": false,
    "SVGAnimatedString": false,
    "SVGAnimatedTransformList": false,
    "SVGAnimateElement": false,
    "SVGAnimateMotionElement": false,
    "SVGAnimateTransformElement": false,
    "SVGAnimationElement": false,
    "SVGCircleElement": false,
    "SVGClipPathElement": false,
    "SVGComponentTransferFunctionElement": false,
    "SVGDefsElement": false,
    "SVGDescElement": false,
    "SVGDiscardElement": false,
    "SVGElement": false,
    "SVGEllipseElement": false,
    "SVGFEBlendElement": false,
    "SVGFEColorMatrixElement": false,
    "SVGFEComponentTransferElement": false,
    "SVGFECompositeElement": false,
    "SVGFEConvolveMatrixElement": false,
    "SVGFEDiffuseLightingElement": false,
    "SVGFEDisplacementMapElement": false,
    "SVGFEDistantLightElement": false,
    "SVGFEDropShadowElement": false,
    "SVGFEFloodElement": false,
    "SVGFEFuncAElement": false,
    "SVGFEFuncBElement": false,
    "SVGFEFuncGElement": false,
    "SVGFEFuncRElement": false,
    "SVGFEGaussianBlurElement": false,
    "SVGFEImageElement": false,
    "SVGFEMergeElement": false,
    "SVGFEMergeNodeElement": false,
    "SVGFEMorphologyElement": false,
    "SVGFEOffsetElement": false,
    "SVGFEPointLightElement": false,
    "SVGFESpecularLightingElement": false,
    "SVGFESpotLightElement": false,
    "SVGFETileElement": false,
    "SVGFETurbulenceElement": false,
    "SVGFilterElement": false,
    "SVGForeignObjectElement": false,
    "SVGGElement": false,
    "SVGGeometryElement": false,
    "SVGGradientElement": false,
    "SVGGraphicsElement": false,
    "SVGImageElement": false,
    "SVGLength": false,
    "SVGLengthList": false,
    "SVGLinearGradientElement": false,
    "SVGLineElement": false,
    "SVGMarkerElement": false,
    "SVGMaskElement": false,
    "SVGMatrix": false,
    "SVGMetadataElement": false,
    "SVGMPathElement": false,
    "SVGNumber": false,
    "SVGNumberList": false,
    "SVGPathElement": false,
    "SVGPatternElement": false,
    "SVGPoint": false,
    "SVGPointList": false,
    "SVGPolygonElement": false,
    "SVGPolylineElement": false,
    "SVGPreserveAspectRatio": false,
    "SVGRadialGradientElement": false,
    "SVGRect": false,
    "SVGRectElement": false,
    "SVGScriptElement": false,
    "SVGSetElement": false,
    "SVGStopElement": false,
    "SVGStringList": false,
    "SVGStyleElement": false,
    "SVGSVGElement": false,
    "SVGSwitchElement": false,
    "SVGSymbolElement": false,
    "SVGTextContentElement": false,
    "SVGTextElement": false,
    "SVGTextPathElement": false,
    "SVGTextPositioningElement": false,
    "SVGTitleElement": false,
    "SVGTransform": false,
    "SVGTransformList": false,
    "SVGTSpanElement": false,
    "SVGUnitTypes": false,
    "SVGUseElement": false,
    "SVGViewElement": false,
    "TaskAttributionTiming": false,
    "Text": false,
    "TextDecoder": false,
    "TextEncoder": false,
    "TextEvent": false,
    "TextMetrics": false,
    "TextTrack": false,
    "TextTrackCue": false,
    "TextTrackCueList": false,
    "TextTrackList": false,
    "TimeRanges": false,
    "toolbar": false,
    "top": false,
    "Touch": false,
    "TouchEvent": false,
    "TouchList": false,
    "TrackEvent": false,
    "TransitionEvent": false,
    "TreeWalker": false,
    "UIEvent": false,
    "URL": false,
    "URLSearchParams": false,
    "ValidityState": false,
    "visualViewport": false,
    "VisualViewport": false,
    "VTTCue": false,
    "WaveShaperNode": false,
    "WebAssembly": false,
    "WebGL2RenderingContext": false,
    "WebGLActiveInfo": false,
    "WebGLBuffer": false,
    "WebGLContextEvent": false,
    "WebGLFramebuffer": false,
    "WebGLProgram": false,
    "WebGLQuery": false,
    "WebGLRenderbuffer": false,
    "WebGLRenderingContext": false,
    "WebGLSampler": false,
    "WebGLShader": false,
    "WebGLShaderPrecisionFormat": false,
    "WebGLSync": false,
    "WebGLTexture": false,
    "WebGLTransformFeedback": false,
    "WebGLUniformLocation": false,
    "WebGLVertexArrayObject": false,
    "WebSocket": false,
    "WheelEvent": false,
    "window": false,
    "Window": false,
    "Worker": false,
    "WritableStream": false,
    "XMLDocument": false,
    "XMLHttpRequest": false,
    "XMLHttpRequestEventTarget": false,
    "XMLHttpRequestUpload": false,
    "XMLSerializer": false,
    "XPathEvaluator": false,
    "XPathExpression": false,
    "XPathResult": false,
    "XSLTProcessor": false,
    "exports": true,
    "global": false,
    "module": false,
    "require": false,
    "Array": false,
    "ArrayBuffer": false,
    "Boolean": false,
    "constructor": false,
    "DataView": false,
    "Date": false,
    "decodeURI": false,
    "decodeURIComponent": false,
    "encodeURI": false,
    "encodeURIComponent": false,
    "Error": false,
    "escape": false,
    "eval": false,
    "EvalError": false,
    "Float32Array": false,
    "Float64Array": false,
    "Function": false,
    "hasOwnProperty": false,
    "Infinity": false,
    "Int16Array": false,
    "Int32Array": false,
    "Int8Array": false,
    "isFinite": false,
    "isNaN": false,
    "isPrototypeOf": false,
    "JSON": false,
    "Map": false,
    "Math": false,
    "NaN": false,
    "Number": false,
    "Object": false,
    "parseFloat": false,
    "parseInt": false,
    "Promise": false,
    "propertyIsEnumerable": false,
    "Proxy": false,
    "RangeError": false,
    "ReferenceError": false,
    "Reflect": false,
    "RegExp": false,
    "Set": false,
    "String": false,
    "Symbol": false,
    "SyntaxError": false,
    "toLocaleString": false,
    "toString": false,
    "TypeError": false,
    "Uint16Array": false,
    "Uint32Array": false,
    "Uint8Array": false,
    "Uint8ClampedArray": false,
    "undefined": false,
    "unescape": false,
    "URIError": false,
    "valueOf": false,
    "WeakMap": false,
    "WeakSet": false,
    "afterAll": false,
    "afterEach": false,
    "beforeAll": false,
    "beforeEach": false,
    "describe": false,
    "expect": false,
    "fdescribe": false,
    "fit": false,
    "it": false,
    "jest": false,
    "pit": false,
    "test": false,
    "xdescribe": false,
    "xit": false,
    "xtest": false,
    "__dirname": false,
    "__filename": false,
    "Buffer": false,
    "clearImmediate": false,
    "process": false,
    "setImmediate": false
  },
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "jest": true,
    "node": true
  },
  "rules": {
    "array-callback-return": "warn",
    "default-case": [
      "warn",
      {
        "commentPattern": "^no default$"
      }
    ],
    "dot-location": [
      "warn",
      "property"
    ],
    "eqeqeq": [
      "warn",
      "smart"
    ],
    "new-parens": "warn",
    "no-array-constructor": "warn",
    "no-caller": "warn",
    "no-cond-assign": [
      "warn",
      "except-parens"
    ],
    "no-const-assign": "warn",
    "no-control-regex": "warn",
    "no-delete-var": "warn",
    "no-dupe-args": "warn",
    "no-dupe-class-members": "warn",
    "no-dupe-keys": "warn",
    "no-duplicate-case": "warn",
    "no-empty-character-class": "warn",
    "no-empty-pattern": "warn",
    "no-eval": "warn",
    "no-ex-assign": "warn",
    "no-extend-native": "warn",
    "no-extra-bind": "warn",
    "no-extra-label": "warn",
    "no-fallthrough": "warn",
    "no-func-assign": "warn",
    "no-implied-eval": "warn",
    "no-invalid-regexp": "warn",
    "no-iterator": "warn",
    "no-label-var": "warn",
    "no-labels": [
      "warn",
      {
        "allowLoop": true,
        "allowSwitch": false
      }
    ],
    "no-lone-blocks": "warn",
    "no-loop-func": "warn",
    "no-mixed-operators": [
      "warn",
      {
        "groups": [
          [
            "&",
            "|",
            "^",
            "~",
            "<<",
            ">>",
            ">>>"
          ],
          [
            "==",
            "!=",
            "===",
            "!==",
            ">",
            ">=",
            "<",
            "<="
          ],
          [
            "&&",
            "||"
          ],
          [
            "in",
            "instanceof"
          ]
        ],
        "allowSamePrecedence": false
      }
    ],
    "no-multi-str": "warn",
    "no-native-reassign": "warn",
    "no-negated-in-lhs": "warn",
    "no-new-func": "warn",
    "no-new-object": "warn",
    "no-new-symbol": "warn",
    "no-new-wrappers": "warn",
    "no-obj-calls": "warn",
    "no-octal": "warn",
    "no-octal-escape": "warn",
    "no-redeclare": "warn",
    "no-regex-spaces": "warn",
    "no-restricted-syntax": [
      "warn",
      "WithStatement"
    ],
    "no-script-url": "warn",
    "no-self-assign": "warn",
    "no-self-compare": "warn",
    "no-sequences": "warn",
    "no-shadow-restricted-names": "warn",
    "no-sparse-arrays": "warn",
    "no-template-curly-in-string": "warn",
    "no-this-before-super": "warn",
    "no-throw-literal": "warn",
    "no-undef": "error",
    "no-restricted-globals": [
      "error",
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "defaultstatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top"
    ],
    "no-unexpected-multiline": "warn",
    "no-unreachable": "warn",
    "no-unused-expressions": [
      "error",
      {
        "allowShortCircuit": true,
        "allowTernary": true,
        "allowTaggedTemplates": true
      }
    ],
    "no-unused-labels": "warn",
    "no-unused-vars": [
      "warn",
      {
        "args": "none",
        "ignoreRestSiblings": true
      }
    ],
    "no-use-before-define": [
      "warn",
      {
        "functions": false,
        "classes": false,
        "variables": false
      }
    ],
    "no-useless-computed-key": "warn",
    "no-useless-concat": "warn",
    "no-useless-constructor": "warn",
    "no-useless-escape": "warn",
    "no-useless-rename": [
      "warn",
      {
        "ignoreDestructuring": false,
        "ignoreImport": false,
        "ignoreExport": false
      }
    ],
    "no-with": "warn",
    "no-whitespace-before-property": "warn",
    "require-yield": "warn",
    "rest-spread-spacing": [
      "warn",
      "never"
    ],
    "strict": [
      "warn",
      "never"
    ],
    "unicode-bom": [
      "warn",
      "never"
    ],
    "use-isnan": "warn",
    "valid-typeof": "warn",
    "no-restricted-properties": [
      "error",
      {
        "object": "require",
        "property": "ensure",
        "message": "Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting"
      },
      {
        "object": "System",
        "property": "import",
        "message": "Please use import() instead. More info: https://facebook.github.io/create-react-app/docs/code-splitting"
      }
    ],
    "getter-return": "warn",
    "import/first": "error",
    "import/no-amd": "error",
    "import/no-webpack-loader-syntax": "error",
    "react/forbid-foreign-prop-types": [
      "warn",
      {
        "allowInPropTypes": true
      }
    ],
    "react/jsx-no-comment-textnodes": "warn",
    "react/jsx-no-duplicate-props": [
      "warn",
      {
        "ignoreCase": true
      }
    ],
    "react/jsx-no-target-blank": "warn",
    "react/jsx-no-undef": "error",
    "react/jsx-pascal-case": [
      "warn",
      {
        "allowAllCaps": true,
        "ignore": []
      }
    ],
    "react/jsx-uses-react": "warn",
    "react/jsx-uses-vars": "warn",
    "react/no-danger-with-children": "warn",
    "react/no-direct-mutation-state": "warn",
    "react/no-is-mounted": "warn",
    "react/no-typos": "error",
    "react/react-in-jsx-scope": "error",
    "react/require-render-return": "error",
    "react/style-prop-object": "warn",
    "jsx-a11y/accessible-emoji": "warn",
    "jsx-a11y/alt-text": "warn",
    "jsx-a11y/anchor-has-content": "warn",
    "jsx-a11y/anchor-is-valid": [
      "warn",
      {
        "aspects": [
          "noHref",
          "invalidHref"
        ]
      }
    ],
    "jsx-a11y/aria-activedescendant-has-tabindex": "warn",
    "jsx-a11y/aria-props": "warn",
    "jsx-a11y/aria-proptypes": "warn",
    "jsx-a11y/aria-role": "warn",
    "jsx-a11y/aria-unsupported-elements": "warn",
    "jsx-a11y/heading-has-content": "warn",
    "jsx-a11y/iframe-has-title": "warn",
    "jsx-a11y/img-redundant-alt": "warn",
    "jsx-a11y/no-access-key": "warn",
    "jsx-a11y/no-distracting-elements": "warn",
    "jsx-a11y/no-redundant-roles": "warn",
    "jsx-a11y/role-has-required-aria-props": "warn",
    "jsx-a11y/role-supports-aria-props": "warn",
    "jsx-a11y/scope": "warn",
    "flowtype/define-flow-type": "warn",
    "flowtype/require-valid-file-annotation": "warn",
    "flowtype/use-flow-type": "warn"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "globalReturn": true,
      "jsx": true
    },
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "root": true,
  "parser": "d:\\web\\project-folder\\front\\node_modules\\babel-eslint\\lib\\index.js",
  "plugins": [
    "import",
    "flowtype",
    "jsx-a11y",
    "react"
  ],
  "settings": {
    "react": {
      "version": "detect"
    }
  },
  "extends": "react-app"
}
Done in 0.78s.

```

**Comment by jazzfog** [comment]
Actually I see another issue:
```js
const x: any = 'Hello';
const y: string = x as string;
```
Results in
```
d:\web\kino\front\src\Button.tsx
  10:21  error  Parsing error: Unexpected token, expected ";"

   8 |
   9 | const x: any = 'Hello';
> 10 | const y: string = x as string;
     |                     ^
```

**Comment by not-an-aardvark** [comment]
What version of the `react-scripts` package are you using? It seems like the `babel-eslint` parser is getting used for your `tsx` file, but I would have expected the `typescript-eslint/parser` parser to be used instead based on [this](https://github.com/facebook/create-react-app/blob/1a61db58d434d33603f20e73ca643ec83c561b73/packages/eslint-config-react-app/index.js#L57). (I'm not sure about whether `babel-eslint` tries to support TypeScript syntax.)

**Comment by jazzfog** [comment]
This is what I have in `package.json` generated by `create-react-app`
```
"react-scripts": "2.1.8",
```
Let me know If you need more information.
Thanks for looking into it!
